state-sync/aptos-data-client/src/latency_monitor.rs

```diff
@@ -83,6 +83,28 @@ impl LatencyMonitor {
                 },
             };
 
+            // Get the latest block timestamp from storage
+            let latest_block_timestamp_usecs = match self
+                .storage
+                .get_block_timestamp(highest_synced_version)
+            {
+                Ok(block_timestamp_usecs) => block_timestamp_usecs,
+                Err(error) => {
+                    sample!(
+                        SampleRate::Duration(Duration::from_secs(LATENCY_MONITOR_LOG_FREQ_SECS)),
+                        warn!(
+                            (LogSchema::new(LogEntry::LatencyMonitor)
+                                .event(LogEvent::StorageReadFailed)
+                                .message(&format!("Unable to read the latest block timestamp: {:?}", error)))
+                        );
+                    );
+                    continue; // Continue to the next round
```

Comment:
it's kinda weird that each clause has the option of skipping the later clauses; I guess they're not independent and if one fails the others are likely to fail?

Reply:
Yeah, they're all dependent. So, I'm not super worried. The only way this fails is if storage fails, and if that happens, we're cooked anyway 😄